### PR TITLE
TIG-4098: Move FTDC decoder code to a new public repo and adding tests

### PR DESCRIPTION
**Pull request by mridul-augustine**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
+name = "ftdc-tools"
+version = "0.1.0"
+description = ""
+authors = ["Mridul Augustine <mridul.augustine@mongodb.com>"]
+
+[tool.poetry.dependencies]
```
Comment (aggrand) [inline pyproject.toml:7]: Should we add linters, like flake8 and mypy?

E.g. [here](https://github.com/10gen/dev-prod-etls/blob/9a1252b90356fb3c7598f766b1089ecdf2c33951/images/calculate-rollups/pyproject.toml#L15) are all the linters we have set up for the argo images.

File: ftdc_tools/ftdc_decoder.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
+import collections
```
Comment (aggrand) [inline ftdc_tools/ftdc_decoder.py:2]: I wasn't on the original review, so I'm assuming most of this was unchanged and I shouldn't look at it? Please lmk if that's wrong and I can give it a review as well.

File: README.md [hunk 3]
```diff
@@ -1 +1,75 @@
...
+* This a pure python package and does not rely on any external binaries for decoding FTDC data.
+* This package provides streaming support. With this feature the whole FTDC files is not loaded in memory and is capable of processing large FTDC file.
+* Async support for faster processing.
+
+## Getting Started - Users
+```
+pip install ftdc-tools
+```
+
+## Usage
```
Comment (aggrand) [inline README.md:13]: Can these examples explain the `memory` parameter and how to use it?

File: tests/test_ftdc_decode.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import pytest
+from aiofile import async_open
+
+from ftdc_tools.ftdc_decoder import FTDC
+
+
+def validateFTDCRecord(record):
```
Comment (aggrand) [inline tests/test_ftdc_decode.py]: So most of what we're validating is the well-structuredness of the FTDC records? Should we add tests for the case where values are something other than what we expected? e.g. what if `ops` is negative? Can that happen somehow? Can we handle it if so? (Presumably via an error?) Also, what happens if the file in question isn't an FTDC file?

I know you'd mentioned stuff being hard without an encoding lib, so defs push back if there's not a good way to do that kind of testing. And our integration testing of the rollup code might be sufficient.

**Review by aggrand — CHANGES_REQUESTED** [review]
Had a few questions, but this is looking good!

Were we intending to publish this as a pip-installable package as part of this ticket? Same with the rollup code? I was hoping to pip-install both of them into the argo images after this ticket, but if that's something we should do separately lmk and I can cut a separate ticket for that!

File: ftdc_tools/ftdc_decoder.py (hunk 2 above)
Comment (mridul-augustine) [reply ftdc_tools/ftdc_decoder.py:2]: Yes nothing changed here.

File: tests/test_ftdc_decode.py (hunk 4 above)
Comment (mridul-augustine) [reply tests/test_ftdc_decode.py]: yes, it's hard without encoding lib. Also, IMO the FTDC decoding library should be generic and shouldn't worry about what columns or values they can have. Let me know if you disagree.

**Review by aggrand — APPROVED** [review]
LGTM! Consider setting up CI to run the tests, either using evergreen (which would take longer) or maybe a mini drone pipeline with a step to run the tests and lint. The rest of everything looks good though!

**Pull request merged**